File: python/2420找到所有好下标.py
def goodIndices(nums: list, k: int) -> list:
    size=len(nums)
    left_dp=[1]*size
    right_dp=[1]*size

    for i in range(1,size):
        if(nums[i]<=nums[i-1]):
            left_dp[i]+=left_dp[i-1]
    for i in range(1,size):
        if(nums[i]>=nums[i-1]):
            right_dp[i]+=right_dp[i-1]
            
    print(left_dp)
    print(right_dp)
    return [i for i in range(k,size-k) if left_dp[i-1]>=k and right_dp[i+k]>=k]

File: cpp/2420找到所有好下标.cpp
#include<vector>

using namespace std;

vector<int> goodIndices(vector<int> &nums, int k)
{
    vector<int> res;
    const int len = nums.size();
    int left_dp[len];
    int right_dp[len];

    for (int i = 0; i < len; i++)
        left_dp[i] = right_dp[i] = 1;

    for (int i = 1; i < len; i++)
    {
        if(nums[i]<=nums[i-1])
            left_dp[i] += left_dp[i - 1];
    }
    for (int i = 1; i < len; i++)
    {
        if(nums[i]>=nums[i-1])
            right_dp[i] += right_dp[i - 1];
    }

    for (int i = k; i < len-k; i++)
    {
        if (left_dp[i - 1] >= k && right_dp[i + k] >= k)
            res.push_back(i);
    }

    return res;
}